google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelector.java

```diff
@@ -267,12 +269,13 @@ public Long getProjectNumber() {
    */
   @Nullable
   public Project getProject() {
-    ResourceProjectModelItem modelItem = getCurrentModelItem();
-    return modelItem != null ? modelItem.getProject() : null;
+   ModelItem modelItem = getCurrentModelItem();
+   return modelItem instanceof ResourceProjectModelItem ?
+       ((ResourceProjectModelItem)modelItem).getProject() : null;
   }
 
   @Nullable
-  private ResourceProjectModelItem getCurrentModelItem() {
+  private ModelItem getCurrentModelItem() {
     if (Strings.isNullOrEmpty(getText())) {
       return null;
```

Comment:
Its a bit confusing that you are using both null and `InvalidProjectModelItem`. It looks like if the selector is empty (no text) you return null, but if the selector has text but is not an existing project you return InvalidProjectModelItem. Why the need for both?

Reply:
Oh I think I get it. Its because you specifically need to handle this null case (where there is no text) differently than an "invalid" project.